packages/@aws-cdk/aws-events-targets/lib/util.ts

```diff
@@ -45,3 +46,36 @@ export function addLambdaPermission(rule: events.IRule, handler: lambda.IFunctio
     });
   }
 }
+
+
+export function addToDeadLetterQueueResourcePolicy(rule: events.IRule, queue: sqs.IQueue) {
+  const ruleParsedStack = Stack.of(rule);
+  const queueParsedStack = Stack.of(queue);
+
+  if (ruleParsedStack.region !== queueParsedStack.region) {
+    throw new Error(`Cannot assign Dead Letter Queue to the rule ${rule}. Both the queue and the rule must be in the same region`);
+  }
+
+  // Skip Resource Policy creation if the Queue is not in the same account.
+  // There is no way to add a target onto an imported rule, so we can assume we will run the following code only
+  // in the account where the rule is created.
+
+  if (ruleParsedStack.account === queueParsedStack.account) {
+    const policyStatementId = `AllowEventRule${Names.nodeUniqueId(rule.node)}`;
+
+    queue.addToResourcePolicy(new iam.PolicyStatement({
+      sid: policyStatementId,
+      principals: [new iam.ServicePrincipal('events.amazonaws.com')],
+      effect: iam.Effect.ALLOW,
+      actions: ['sqs:SendMessage'],
+      resources: [queue.queueArn],
+      conditions: {
+        ArnEquals: {
+          'aws:SourceArn': rule.ruleArn,
+        },
+      },
+    }));
+  } else {
+    // Maybe we could post a warning telling the user to create the permission in the target account manually ?
```

Comment:
Should we add a warning in case the queue is from another account and we cannot add the resource policy ?